R/plot_cbass.R

```diff
@@ -906,30 +1191,10 @@ cbass_dendro_plotly <- function(x,
     if(show_clusters){
       labels <- get_cluster_labels(x, k.row = k.row, k.col = k.col, percent = percent, type = type)
       k <- nlevels(labels)
-      if(has_percent){
-        if (!is_percent_scalar(percent)) {
-          crv_error(sQuote("percent"), " must be a scalar between 0 and 1 (inclusive).")
-        }
-
-        # if(percent == 0){ ## Don't bother showing boxes if percent is 0 and bail early
-        #   return(invisible(x))
-        # }
-      } else {
-        if (!is_integer_scalar(k)) {
-          crv_error(sQuote("k"), " must be an integer scalar (vector of length 1).")
-        }
-        if ( k <= 0 ) {
-          crv_error(sQuote("k"), " must be positive.")
-        }
-        if ( k > NROW(x$X) ) {
-          crv_error(sQuote("k"), " cannot be more than the observations in the original data set (", NROW(x$X), ").")
-        }
-
-        percent <- x$row_fusions$cluster_membership %>%
-          select(.data$GammaPercent, .data$NCluster) %>%
-          filter(.data$NCluster <= k) %>%
+      if(!has_percent){
+        percent <- get_feature_paths(x, features = character(), type = type) %>% filter(.data$NCluster == k) %>%
           select(.data$GammaPercent) %>%
-          summarize(percent = min(.data$GammaPercent)) %>%
+          summarize(NCluster = mean(.data$GammaPercent)) %>%
```

Comment:
I'm confused by this line - why is the number of clusters equal to the average value of $\lambda$? This are very different things...